### DISCUSSION TITLE
OIDC PKCE support

### DISCUSSION THREAD

**Discussion by D3an1el** [minio/minio#20239] [Q&A]
Hi, i recently was trying too get minio oidc to work with kanidm. With Kanidm's strict security with oauth2 it requires PKCE . Is this going to be implemented ? Please see link for reasoning.

Reason for PKCE:

https://kanidm.github.io/kanidm/master/frequently_asked_questions.html#oauth2)

https://oauth.net/2/pkce/

**Comment by asep-nugraha** [comment]
Yes please I agree, I need the minio can authorize using authorization code authentication method, using PKCE enabled.
Because this is the best practice using PKCE enabled when using authorization code authentication on OIDC Provider, it can close the security hole. I am using Identity Server 4.0 for OIDC Provider.

https://identityserver4.readthedocs.io/en/latest/
